src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -14,6 +14,11 @@ namespace NuGet.ProjectModel
 {
     public class DependencyGraphSpec
     {
+        /// <summary>
+        /// Allows a user to enable the legacy SHA512 hash function for dgSpec files which is used by no-op.
+        /// </summary>
+        private static readonly bool UseLegacyHashFunction = string.Equals(Environment.GetEnvironmentVariable("NUGET_ENABLE_LEGACY_DGSPEC_HASH_FUNCTION"), bool.TrueString, StringComparison.OrdinalIgnoreCase);
```

Comment:
This can be done later, but we tend to prefer EnvironmentVariableReaderWrapper for checks like this.

Reply:
Yeah I wasn't going to bother with unit tests so for now I just directly read the value.  If needed, I can change it later on as you mentioned.